src/go/rdctl/cmd/extensionInstall.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright © 2023 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cmd implements the rdctl commands
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+)
+
+// installCmd represents the 'rdctl extensions install' command
+var installCmd = &cobra.Command{
+	Use:   "install",
+	Short: "Install an RDX extension",
+	Long: `rdctl extension install [--force] <image-id>
+--force: avoid any interactivity.
+The <image-id> is an image reference, e.g. splatform/epinio-docker-desktop:latest (the tag is optional).`,
+	Args: cobra.ExactArgs(1),
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		return installExtension(args)
+	},
+}
+
+func init() {
+	extensionCmd.AddCommand(installCmd)
+}
+
+func installExtension(args []string) error {
+	imageID := args[0]
+	endpoint := fmt.Sprintf("/%s/extensions/install?id=%s", apiVersion, imageID)
+	// https://stackoverflow.com/questions/20847357/golang-http-client-always-escaped-the-url
+	// Looks like http.NewRequest(method, url) escapes the URL
+
+	result, errorPacket, err := processRequestForAPI(doRequest("POST", endpoint))
+	if errorPacket != nil || err != nil {
+		if result != nil {
+			return fmt.Errorf("installation failed: %s\n", string(result))
+		}
```

Comment:
It feels weird that for "install" the `result` of a failed call is expected/meaningful and will be displayed as the error, but for "ls" and "uninstall" the `result` is ignored in case of an error.

Seems like an inconsistent API design.

Reply:
So when you pass an argument to `uninstall`, it almost always succeeds:
```
$ rdctl extension uninstall 'akkk!!flipp!'
Uninstalling image akkk!!flipp!: Deleted akkk!!flipp!
$ rdctl extension uninstall 'akkk!! flipp!'
{"message":"400 Bad Request"}
$ rdctl extension uninstall ''
{"message":"400 Bad Request"}
```

I think this PR is ok. At some point we can nail down what the `rdctl` output should be more systematically.

Reply:
So you are saying it is the API that is behaving erratically when you issue an "uninstall" request?